ofrak_core/test_ofrak/unit/test_ofrak_server.py

```diff
@@ -30,6 +29,13 @@ def hello_world_elf() -> bytes:
     return hello_elf()
 
 
+@pytest.fixture()
+async def test_project_dir(ofrak_client: TestClient):
+    with tempfile.TemporaryDirectory() as tempdir:
+        await ofrak_client.post("/set_projects_path", json={"path": tempdir})
+        yield tempdir
+
+
```

Comment:
A more idiomatic way to do this would be to use pytest's builtin `tempdir` fixture: https://docs.pytest.org/en/6.2.x/tmpdir.html.

It would be something like this:

```
@pytest.fixture()
async def project_dir(ofrak_client: TestClient, tmpdir):
    await ofrak_client.post("/set_projects_path", json={"path": tempdir.path}
    yield tempdir.path
```

Note that we don't want to start this function with `test_` since it is not a test (this follows pytest naming conventions).

Reply:
You also might want to make this an "autouse" fixture so it does not need to be invoked for each test (https://docs.pytest.org/en/6.2.x/fixture.html?highlight=autouse).

A clean way to do this would be to create a "TestOfrakProject" class, add that fixture to the class, and make all of the tests methods on that class.

Reply:
There are already projects tests in https://github.com/redballoonsecurity/ofrak/tree/1653a17a722a45a7a64bc717ce1196059cd8a7c7/ofrak_core/test_ofrak/unit/ofrak_project which might get confusing. I like the idea of making classes in this file though, should probably be done across the file if its being done once.

Reply:
Implemented this, turns out if you tag a fixture for `autouse` you can not use the resulting value, even if you pass the fixture into a function explicitly as you would normally. This breaks the `test_get_projects_path` test.

Reply:
Hm. If the autouse doesn't work, I think it still makes sense to use the standard `tmpdir` fixture.